interactions/client.py

```diff
@@ -256,8 +256,8 @@ async def _synchronize(self, payload: Optional[dict] = None) -> None:
                 await self.__create_sync(payload)
         else:
             to_delete.extend(command for command in commands if command not in cache)
-        await self.__bulk_update_sync(to_sync)
-        await self.__bulk_update_sync(to_delete, delete=True)
+        if len(to_sync) >= 1:
+            await self.__bulk_update_sync(to_sync + commands)
```

Comment:
How are you actually communicating that we're deleting? If your goal is only 1 bulk call, I'd also recommend removing the deletion establishment in the bulk update sync method.

Reply:
In the bulk_update_sync() this is the part where the updates to Discord get uploaded 
<img width="638" alt="Snip1" src="https://user-images.githubusercontent.com/51281740/161120251-4ba8ba9f-988a-41f7-bbab-148892ec02b3.png">
in `self.http.overwrite_application_command`
is 
```py
async def overwrite_application_command(
        self, application_id: int, data: List[dict], guild_id: Optional[int] = None
    ) -> List[dict]:
        """
        Overwrites application command(s) from a scope to the new, updated commands.

        ..note:
            This applies to all forms of application commands (slash and context menus)

        :param application_id: Application ID snowflake
        :param data: The dictionary that contains the command (name, description, etc)
        :param guild_id: Guild ID snowflake to put them in, if applicable.
        :return: An array of application command objects.
        """
        url = (
            f"/applications/{application_id}/commands"
            if not guild_id
            else f"/applications/{application_id}/guilds/{guild_id}/commands"
        )

        return await self._req.request(Route("PUT", url), json=data)
```
In Discord API docs  
![image](https://user-images.githubusercontent.com/51281740/161121172-27b9f5dd-2084-4741-8baf-4b4b2c0e461f.png)
and 
![image](https://user-images.githubusercontent.com/51281740/161121237-78d42a7c-c3f3-4314-be12-5943bb96bdd7.png)
which means that if the data given doesn't have those items it will be deleted
Therefore in my code the `to_sync + commands` takes account to all new and old commands together to replace it

Reply:
I don't think you understood what I meant.

In your code, you've stopped using the `delete` kwarg that our internal bulk sync method has for specifically deleting messages. If you're not going to use this, (presuming you're wanting to make only 1 bulk call now) remove the logic from this method including the kwarg allowing this.